77. Combinations/solution.py

```diff
@@ -2,25 +2,31 @@ class Solution:
     # Given two integers n and k, this function returns all possible 
     # combinations of k numbers chosen from the range [1, n].
     def combine(self, n: int, k: int) -> List[List[int]]:
-        # if k is 0, return an empty list.
-
-        # otherwise, return the result of a helper function.
+        # return the result of a helper function.
         return self.rangedCombine(1, n, k)
 
     # Given three integers start, end, and k, this function returns
     # all possible combinations of k numbers chosen from the range
     # [start, end].
     def rangedCombine(self, start: int, end: int, k: int):
         # if k is 0, return a list containing an empty list.
+        if k == 0:
```

Comment:
Seems like you could have had your base case at k == 1 by checking in the for loop but this is pretty clean too. Just one extra level of recursion.